src/csvcubed/models/cube/validationerrors.py

```diff
@@ -92,3 +93,25 @@ def get_error_url(cls) -> str:
     def __post_init__(self):
         row_nums_str = ", ".join([str(i) for i in sorted(self.row_numbers)])
         self.message = f"Missing value(s) found for '{self.csv_column_title}' in row(s) {row_nums_str}."
+
+
+@dataclass
+class UriTemplateNameError(SpecificValidationError):
+    """
+    An error to inform the user that they the column name they are referring to in a csv uri template
+    does not exist.
+    """
+
+    column_names: List[str]
+    csv_column_uri_template: str
+    column_names_concatenated: Optional[str] = None
```

Comment:
Since this is being populated in the `__post_init__` method, ` = field(init=False)` is the best thing to stick here since it makes sure it doesn't appear in the constructor/`__init__` at all.